src/kernel/src/fs/file.rs

```diff
@@ -75,8 +79,25 @@ impl Write for VFile {
 
 /// Type of [`VFile`].
 #[derive(Debug)]
+#[rustfmt::skip]
 pub enum VFileType {
-    Vnode(Arc<Vnode>), // DTYPE_VNODE
+    Vnode(Arc<Vnode>), // DTYPE_VNODE = 1
+//  Socket,            // DTYPE_SOCKET = 2
+//  Pipe,              // DTYPE_PIPE = 3
+//  Fifo,              // DTYPE_FIFO = 4
+//  Kqueue,            // DTYPE_KQUEUE = 5
+//  Crypto,            // DTYPE_CRYPTO = 6
+//  Mqueue,            // DTYPE_MQUEUE = 7 (POSIX message queues)
+//  Shm,               // DTYPE_SHM = 8 (POSIX shared memory)
+//  Sem,               // DTYPE_SEM = 9 (POSIX semaphores)
+//  Pts,               // DTYPE_PTS = 10
+//  Dev,               // DTYPE_DEV = 11
+//  Cap,               // DTYPE_CAPABILITY = 12
+//  ProcDesc,          // DTYPE_PROCDESC = 13
+//  JitShm,            // (presumably) DTYPE_JITSHM = 14
+//  Socket2,           // unknown = 15 // figure out what exactly this is
```

Comment:
@red-prig Do you know what this is?

Reply:
@SuchAFuriousDeath These are the internal file type identifiers, for PS4 they are:

```
 DTYPE_VNODE     = 1; { file }
 DTYPE_SOCKET    = 2; { communications endpoint }
 DTYPE_PIPE      = 3; { pipe }
 DTYPE_FIFO      = 4; { fifo (named pipe) }
 DTYPE_KQUEUE    = 5; { event queue }
 DTYPE_CRYPTO    = 6; { crypto }
 DTYPE_MQUEUE    = 7; { posix message queue }
 DTYPE_SHM       = 8; { swap-backed shared memory }
 DTYPE_SEM       = 9; { posix semaphore }
 DTYPE_PTS       =10; { pseudo teletype master device }
 DTYPE_DEV       =11; { Device specific fd type }
 DTYPE_CAPABILITY=12; { capability }
 DTYPE_PROCDESC  =13; { process descriptor }
 DTYPE_JITSHM    =14; { shared memory for JIT compiler }
 DTYPE_IPCSOCKET =15; { unix domain socket }
 DTYPE_PHYSHM    =16; { PS4 physical shared memory }
 DTYPE_BLOCKPOOL =17; { PlayStation blockpool }
```